subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -371,6 +578,10 @@ static void gen_prov_cont(struct prov_rx *rx, struct net_buf_simple *buf)
 		}
 
 		return;
+	} else if (rx->xact_id == next_transaction_id(link.rx.id) && link.tx.adv[0]) {
```

Comment:
This can happend because the adv is delayed and the link can be closed down.
This was needed when having prov_clear_tx in gen_prov_start and gen_prov_cont which clear the tx buffer, and then the message waiting to go out was failing. This lead to that the callback in gen_prov_ack was not called and remote provisioning failed in sending provisioning start because of the prov_clear_tx prevented the callback in gen_prov_ack to be called. This was needed to make the tx_complete to be called in the rpr_srv.

Reply:
Could you add this comment here? This is quite rare case that requires deep understanding of it.